src/metabase/models/dashboard.clj

```diff
@@ -34,20 +34,31 @@
   {:hydrate :ordered_cards}
   [dashboard-or-id]
   (db/do-post-select DashboardCard
-    (db/query {:select    [:dashcard.*]
+    (db/query {:select    [:dashcard.* [:collection.type :collection_type]]
                :from      [[DashboardCard :dashcard]]
-               :left-join [[Card :card] [:= :dashcard.card_id :card.id]]
+               :left-join [[Card :card] [:= :dashcard.card_id :card.id]
+                           [Collection :collection] [:= :collection.id :card.collection_id]]
                :where     [:and
                            [:= :dashcard.dashboard_id (u/the-id dashboard-or-id)]
                            [:or
                             [:= :card.archived false]
                             [:= :card.archived nil]]] ; e.g. DashCards with no corresponding Card, e.g. text Cards
                :order-by  [[:dashcard.created_at :asc]]})))
 
+(defn collection-type
+  "Returns the collection type associated with the dashboard"
+  {:hydrate :collection_type}
+  [dashboard-or-id]
+  (-> (db/query {:select    [[:collection.type :collection_type]]
+                 :from      [[:report_dashboard :dashboard]]
+                 :left-join [[Collection :collection] [:= :collection.id :dashboard.collection_id]]
+                 :where     [:= :dashboard.id (u/the-id dashboard-or-id)]})
+      first
+      :collection_type))
```

Comment:
I think you should make this a `:batched-hydrate` function. If you fetch multiple Dashboards and hydrate `:collection_type` it will have to do a DB call for each Dashboard and we'd end up with an N+1 problem. This would be simple enough to do in a single request tho.

Toucan hydration can still use the `:batched-hydrate` functions to hydrate a single object